core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -186,7 +186,7 @@ private[kafka] object LogValidator {
       if (record.magic != messageFormatVersion)
         inPlaceAssignment = false
 
-      validatedRecords += record.convert(messageFormatVersion)
+      validatedRecords += record.convert(messageFormatVersion, messageTimestampType)
```

Comment:
Looked into this. Our config definition constrains message.timestamp.type to be one of "CreateTime" or "LogAppendTime," so I think we are safe.